paperqa/llms.py

```diff
@@ -903,12 +911,138 @@ async def similarity_search(
         # we could use arg-partition here
         # but a lot of algorithms expect a sorted list
         sorted_indices = np.argsort(-similarity_scores)
+
         return (
             [self.texts[i] for i in sorted_indices[:k]],
             [similarity_scores[i] for i in sorted_indices[:k]],
         )
 
 
+class QdrantVectorStore(VectorStore):
+    client: Any = Field(
+        default=None,
+        description="Instance of `qdrant_client.QdrantClient`. Defaults to an in-memory instance.",
+    )
+    collection_name: str = Field(default_factory=lambda: f"paper-qa-{uuid.uuid4().hex}")
+    vector_name: str | None = Field(default=None)
+    _ids: list[str] = []
```

Comment:
This exposes to mutable default issues, can you use `Field(default_factory=list)`

Also, should this be a `set`/`Collection` as opposed to a `list`? Does order matter?

Why not use `UUID` directly, why make it `str`?

Lastly, can you make the name more verbose, saying what the IDs correspond with?

Reply:
See https://github.com/Future-House/paper-qa/pull/730/commits/8b59afdf8fba71a67d849b5fa21ee468dc652bdc.

> Also, should this be a set/Collection as opposed to a list? Does order matter?

It's a set now.

> This exposes to mutable default issues, can you use Field(default_factory=list)

Field names can't start with underscores as they're private attrs.

> Why not use UUID directly, why make it str?

The client only allows `Union[int, str]` for point Ids.

> Lastly, can you make the name more verbose, saying what the IDs correspond with?

Done

Reply:
Thanks for doing that 👍